### ISSUE TITLE
Aditional Logitech devices

### ISSUE THREAD

**Issue by jhonny-oliveira** [pwr/Solaar#301]
How can I get these Logitech devices (see below) to be detected and managed by Solaar?

I have a keyboard (MK320) connected to the c531 and mouse (G700) connected to the c52e.

dmesg:
```
...
[ 1400.128449] usb 5-2: new full-speed USB device number 6 using ohci-pci
[ 1400.321529] usb 5-2: New USB device found, idVendor=046d, idProduct=c531
[ 1400.321535] usb 5-2: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[ 1400.321538] usb 5-2: Product: USB Receiver
[ 1400.321540] usb 5-2: Manufacturer: Logitech
[ 1400.329960] input: Logitech USB Receiver as /devices/pci0000:00/0000:00:13.0/usb5/5-2/5-2:1.0/0003:046D:C531.0009/input/input29
[ 1400.330128] hid-generic 0003:046D:C531.0009: input,hidraw0: USB HID v1.11 Mouse [Logitech USB Receiver] on usb-0000:00:13.0-2/input0
[ 1400.340789] input: Logitech USB Receiver as /devices/pci0000:00/0000:00:13.0/usb5/5-2/5-2:1.1/0003:046D:C531.000A/input/input30
[ 1400.448717] hid-generic 0003:046D:C531.000A: input,hiddev0,hidraw1: USB HID v1.11 Keyboard [Logitech USB Receiver] on usb-0000:00:13.0-2/input1
[ 1407.370175] usb 5-3: new full-speed USB device number 7 using ohci-pci
[ 1407.563097] usb 5-3: New USB device found, idVendor=046d, idProduct=c52e
[ 1407.563102] usb 5-3: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[ 1407.563105] usb 5-3: Product: USB Receiver
[ 1407.563108] usb 5-3: Manufacturer: Logitech
[ 1407.571530] input: Logitech USB Receiver as /devices/pci0000:00/0000:00:13.0/usb5/5-3/5-3:1.0/0003:046D:C52E.000B/input/input31
[ 1407.630687] hid-generic 0003:046D:C52E.000B: input,hidraw2: USB HID v1.11 Keyboard [Logitech USB Receiver] on usb-0000:00:13.0-3/input0
[ 1407.634899] input: Logitech USB Receiver as /devices/pci0000:00/0000:00:13.0/usb5/5-3/5-3:1.1/0003:046D:C52E.000C/input/input32
[ 1407.694604] hid-generic 0003:046D:C52E.000C: input,hiddev0,hidraw3: USB HID v1.11 Mouse [Logitech USB Receiver] on usb-0000:00:13.0-3/input1
```

$ lsusb -d 046d:
```
Bus 005 Device 007: ID 046d:c52e Logitech, Inc. MK260 Wireless Combo Receiver
Bus 005 Device 006: ID 046d:c531 Logitech, Inc. C-U0007 [Unifying Receiver]
```

**Comment by Lekensteyn** [comment]
See 3ea51fe1a2d75730ec016fda0a51035001483368 for example. Note: maybe someone has also been working on this, so you might want to search the issue tracker first.

**Comment by jrbenito** [comment]
@jhonny-oliveira ,

What kernel and distro are you using? (just to compare with my lsusb here).

Btw, Oliveira seems like Brazilian or Portuguese family name, am I right?

**Comment by jhonny-oliveira** [comment]
Hi!

Following your tip, after some bumps, I made some progress.

This is what I did:
- Downloaded the master zip file from GitHub;
- Made some edits for the MK320 mouse and keyboard combo;
- Made some edits to detect the G700 Gaming Mouse receiver;
- sudo python setup.py build && sudo python setup.py install.

This is what I got:

$ sudo solaar show
Unifying Receiver
  Device path  : /dev/hidraw3
  USB id       : 046d:c52e
  Serial       : EEC09CB6
    Firmware   : 15.00.B0048
  Has 2 paired device(s) out of a maximum of 2.
  Notifications: (none)

  1: Wireless Mouse M215
     Codename     : M215
     Kind         : mouse
     Wireless PID : 1020
     Protocol     : HID++ 1.0
     Polling rate : 8 ms (125Hz)
     Serial number: 4380042E
          Firmware: 19.02.B0032
     The power switch is located on the (unknown).
     Notifications: (none).
     Battery: good, discharging.

  2: Wireless Keyboard MK320
     Codename     : MK320
     Kind         : keyboard
     Wireless PID : 200F
     Protocol     : HID++ 1.0
     Polling rate : 20 ms (50Hz)
     Serial number: 4380042E
          Firmware: 20.00.B0008
     The power switch is located on the (unknown).
     Notifications: (none).
     Battery: full, discharging.

Unifying Receiver
  Device path  : /dev/hidraw1
  USB id       : 046d:c531
  Serial       : 83F892F0
    Firmware   : 21.00.B0007
    Bootloader : 02.14
  Has 1 paired device(s) out of a maximum of 1.
  Notifications: (none)

  1: G700 Gaming Mouse
     Codename     : G700
     Kind         : mouse
     Wireless PID : 1023
     Protocol     : HID++ 1.0
     Polling rate : 8 ms (125Hz)
     Serial number: C07AEF27
          Firmware: 22.02.B0037
        Bootloader: 02.11
             Other: 02.02
     The power switch is located on the base.
     Notifications: (none).
     Battery: good, discharging.

Seems promising, but I'm still facing some issues:
- when running the gui, I get this: "Found a Logitech Receiver
(/dev/hidraw1), but did not have permission to open it. If you've just
installed Solaar, try removing the receiver and plugging it back in.".
Unplugging/Re-plugging does
not make the error go away. Just as a remark, the udev rule file does not
seem to be in place (/lib/udev/rules.d/ or /etc/udev/rules.d/).
--- Shouldn't the setup put the udev rule in place? Is it somewhere else?

- Ubuntu latest version is 9.2, the GitHub version version is also 9.2,
however, the Ubuntu package does not have the patch you pointed out.
--- Can you help pushing it forward?

- Both mouse and keyboard from from MK320 combo have the location as
"unknown". Well, it is also "on the base".
--- Any ideas on how to fix this?

- Once all issues are sorted out, how can I submit my a patches?

- How do I completely remove the manually installed package (installed
files, configuration files, log files, etc...)?

By the way, I'm running Ubuntu 16.10.

Best regards,
Jhonny Oliveira

**Comment by jhonny-oliveira** [comment]
@jrbenito ,

Ubuntu 16.10
Linux dktp 4.8.0-26-generic #28-Ubuntu SMP

PT-PT :-)

**Comment by Lekensteyn** [comment]
If you installed Solaar before, there should be a file in `/lib/udev/rules.d/` or `/etc/udev/rules.d/` which you can replace with the new file.

To send the patch, you have to use git. It is easier than working with zip files and manual patches. There is a full manual (and Getting Started) at https://git-scm.com/doc,

Basically:
 - fork the repo on Github
 - clone your fork
 - commit patch
 - push patch to Github
 - create pull request

Let me know if you get stuck!

**Comment by jhonny-oliveira** [comment]
Done.

Best regards,
Jhonny Oliveira

**Comment by jrbenito** [comment]
Very nice Jhonny, thanks for helping.

Tip: next time, when you commit the message of your Pull Request, add "close #301". By adding "close hash<issue number>"  to a commit message, Github automatically link the PR with the issue and, if merged, automatically closes the issue.

Thanks again.
PS: this relates to PR #302

**Comment by jrbenito** [comment]
Hi @jhonny-oliveira  and @Lekensteyn ,

since this was solved by #302 we can close this ticket. However I don´t have rights for that, could your do this please?
